Refactor arm_mpu to use CMSIS directly

ioannisg opened the pull request:
Initial PR towards implementing #8018.
This PR refactors the ARM MPU implementation to use the CMSIS macro definitions directly.

ioannisg committed:
arch: arm: use CMSIS macro defines for MPU_CTRL register bit setting

This commit removes the macro definitions for MPU_CTRL register
bitmasks, defined in arm_mpu.h, and modifies the MPU driver to
directly use the equivalent macros defined in ARM CMSIS.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
arch: arm: use CMSIS defines for MPU_RBAR register bit setting

This commit removes the macro definitions for MPU_RBAR register
bitmasks, defined in arm_mpu.h, and modifies the MPU driver to
directly use the equivalent macros defined in ARM CMSIS.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
arch: arm: use CMSIS defines for MPU_RASR register bit setting

This commit removes the macro definitions for MPU_RASR register
bitmasks, defined in arm_mpu.h, and modifies the MPU driver to
directly use the equivalent macros defined in ARM CMSIS.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

MaureenHelm commented:
CI is failing because the beetle doesn't have CMSIS headers. You might be able to fix by defining __MPU_PRESENT for that SoC, but I vaguely remember Vincenzo having some issues with that macro when he originally introduced MPU support into Zephyr.

ioannisg committed:
arch: arm: beetle: duplicate CMSIS MPU-related macros

Duplicate the CMSIS MPU-related macro definitions specifically
for Beetle SOC which is is not compliant with ARM CMSIS and
does not include the ARM CMSIS headers.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
@MaureenHelm I added a commit to explicitly define the ARM MPU related macros for Beetle. It looks it is the only arm SOC that does not include CMSIS headers, so this is a straightforward workaround, IMO.
